examples/refinanceFullAfterExtend.ts

```diff
@@ -14,34 +21,38 @@ const emitRefinaceFullAndRepayLoan = async (contract?: Address) => {
   console.log(`offer placed successfully: ${contractVersionString}`);
 
   const emitLoan = await users[1].emitLoan({
-    offer: signedOffer,
+    offerExecution: users[1].offerExecutionFromOffers([signedOffer]),
+    duration: signedOffer.duration,
     tokenId: testTokenId,
   });
-  const { loan } = await emitLoan.waitTxInBlock();
+  const { loan, loanId } = await emitLoan.waitTxInBlock();
   console.log(`loan emitted: ${contractVersionString}`);
 
   const remainingLockup = await users[0].getRemainingLockupSeconds({ loan });
   console.log(`remaining lockup: ${remainingLockup}`);
   await sleep(remainingLockup * 1_000 + SLEEP_BUFFER);
 
-  let refinancedLoan = loan;
-  let refinancedLoanId = 0n;
+  let repayLoan = loan;
+  let repayLoanId = loanId;
 
   try {
     await generateBlock(); // We need to push a new block into the blockchain [anvil issue]
     const { waitTxInBlock } = await users[0].extendLoan({
       loan,
       newDuration: 120n,
-      loanId: loan.source[0].loanId,
+      loanId,
     });
 
     const receipt = await waitTxInBlock();
 
     console.log(`loan extended: ${contractVersionString}`);
 
-    refinancedLoan = receipt.loan;
-    refinancedLoanId = receipt.newLoanId;
+    repayLoan = receipt.loan;
+    repayLoanId = receipt.newLoanId;
 
+    const isV6 =
+      signedOffer.contractAddress === process.env.MULTI_SOURCE_LOAN_CONTRACT_V6 ||
+      !('source' in loan);
```

Comment:
This will always be false

Reply:
I changed it a little bit, but as i'm using methods that return loan from the blockchain, that loan can either have sources or tranches. Then, I can't infer which of the is it and I gotta handle both cases bc of that.

Reply:
Oh, perfect then!